### PR TITLE
Add getStore helper + withStore HoC. Change callback + remove Store + add docs + add tests

### PR DESCRIPTION
**Pull request by aralroca**
Work:

- [x] Implement `getStore` helper
- [x] Implement `withStore` HoC
- [x] Unify callbacks to 1 callback inside the `createStore` **BREAKING CHANGE**.
- [x] Implement callback inside `useStore` and `withStore`
- [x] Remove `Store` component (now with `getStore` and callback change it no longer makes sense) **BREAKING CHANGE**.
- [x] Tests
- [x] [Docs](https://github.com/aralroca/fragstore/blob/c6c69a52c4b82197fd4cd54de31e57de323e1a5e/README.md). 
- [x] now occupies ~~1.2kb~~ ~~1.08kb~~ 984b, should go down to less than 1kb

### CODE REVIEW

**Review by danielart — APPROVED** [review]

**Comment by siddharthborderwala** [comment]
Hi, I am Siddharth and I'd like to mention that this is a great library, it is very simple, straightforward and easy to use. Awesome work 👏

I am using it for my university course project and I was looking for something exactly like getStore. I wanted to know if you have an ETA on how long before this PR will be merged?

Thank you.

**Comment by aralroca** [comment]
@siddharthborderwala Thanks a lot! It will take me a while to document and test everything well and maybe some fix / improvement, maybe 1 or 2 days. But since you show interest I have made a prerelease with this in: 

- `0.6.0-canary.1`

In theory, everything should work fine, but if you encounter any problem you can comment on it here 🙏 😊

It has some breaking changes that need to be better documented. As we are in an early 0.x stage we try to develop fast to make it smaller, easier and more powerful. It will be more and more stable and from version 1.0 it will be 100% semantic versioning.

**Comment by siddharthborderwala** [comment]
Okay, I hear you. Thank you for your time, what I did is use the index.js file inside package directory and converted it to typescript and now I'm using it in my project.

**Comment by aralroca** [comment]
I fix an error on `0.6.0-canary.2`

**Comment by aralroca** [comment]
@danielart the PR is finnaly ready for review.

File: .eslintrc.js [hunk 1]
```diff
@@ -25,6 +25,7 @@ module.exports = {
   'rules': {
     'require-jsdoc': 0,
     'react/prop-types': 0,
+    'prefer-const': 0,
```
Comment (aralroca) [inline .eslintrc.js:28]: @danielart I have seen that Preact in their projects always use let, this makes the final bundle smaller. I have tried it and it does, it reduces the size a little bit.

File: tests/withStore.test.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
+import {Component} from 'react';
+import {render, screen} from '@testing-library/react';
+import {act} from 'react-dom/test-utils';
+import userEvent from '@testing-library/user-event';
+
+import '@babel/polyfill';
+
+import createStore from '../package/index';
+
+describe('useStore', () => {
```
Comment (aralroca) [inline tests/withStore.test.js:10]:
```suggestion
describe('withStore', () => {
```

**Review by danielart — APPROVED** [review]

**Pull request merged**